plugins/vault-backend/src/service/plugin.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+import {
+  VaultApi,
+  VaultExtensionPoint,
+  vaultExtensionPoint,
+} from '@backstage/plugin-vault-node';
+import { VaultBuilder } from './VaultBuilder';
+
+class VaultExtensionPointImpl implements VaultExtensionPoint {
+  vaultClient?: VaultApi;
+  tokenRenewEnabled: boolean = false;
+
+  setClient(vaultClient: VaultApi): void {
+    if (this.vaultClient) {
+      throw new Error('The vault client has been already set');
+    }
+    this.vaultClient = vaultClient;
+  }
+
+  enableTokenRenew(): void {
+    this.tokenRenewEnabled = true;
+  }
+}
+
+/**
+ * Vault backend plugin
+ *
+ * @public
+ */
+export const vaultPlugin = createBackendPlugin({
+  pluginId: 'vault',
+  register(env) {
+    const vaultExtensions = new VaultExtensionPointImpl();
+    env.registerExtensionPoint(vaultExtensionPoint, vaultExtensions);
```

Comment:
See the updated example in [docs](https://backstage.io/docs/backend-system/building-plugins-and-modules/index#extension-points) for how to define extension points without the need of a separate class, it's quite a bit leaner.

Reply:
That indeed looks much simpler. I've updated the file as well